astropy/units/tests/test_units.py

```diff
@@ -148,6 +148,22 @@ def test_represents():
     assert blabla.decompose().bases == [u.s]
 
 
+@pytest.mark.parametrize(
+    "represents, match_",
+    [
+        ("not_a_unit", "did not parse"),
+        ("mmm", "did not parse"),
+        ([5, 6] * u.hr, "more than one element"),
+    ],
+)
+def test_represents_errors(represents, match_):
+    with pytest.raises(ValueError, match=match_):
+        u.Unit("new_unit", represents)
+
+    with pytest.raises(ValueError, match=match_):
+        u.def_unit("new_unit", represents, namespace=locals())
```

Comment:
`namespace` is not required, and if it is omitted then the arguments `u.Unit` is invoked with are exactly the same as the arguments `u.def_unit` is invoked with. That would allow replacing the two `pytest.raises()` calls with a paramaterized test that only has one.

Reply:
OK, makes sense. Done.